pex/pex_rules.bzl

```diff
@@ -82,6 +83,16 @@ def _collect_transitive_reqs(ctx):
   return transitive_reqs
 
 
+def _collect_repos(ctx):
+  repos = {}
+  for dep in ctx.attr.deps:
+    if hasattr(dep.py, "repos"):
+      repos += dep.py.repos
+  for file in egg_file_types.filter(ctx.files.repos):
```

Comment:
Also technically one can specify https URLs pointing at a package repository with `--repos`.  It's not a great practice with regard to Bazel, but I know people will want to do that... Thoughts?